cmd/agent/api/agent/agent.go

```diff
@@ -179,6 +191,60 @@ func getStatus(w http.ResponseWriter, r *http.Request) {
 	w.Write(jsonStats)
 }
 
+func streamLogs(w http.ResponseWriter, r *http.Request) {
+	log.Info("Got a request for stream logs.")
+	w.Header().Set("Transfer-Encoding", "chunked")
+	logMessageReceiver := logs.GetMessageReceiver()
+
+	if logMessageReceiver.IsEnabled() {
+		http.Error(w, "Another client is already streaming logs.", 500)
+		w.(http.Flusher).Flush()
```

Comment:
I found when testing that the error message doesn't make it to the client without flushing. I assume it is because it is a small payload and doesn't fill the buffer up enough to automatically flush? Are you aware of a better way?

Reply:
Sound good.